### PR TITLE
Make cephcsi controller server and node server configurable

### PR DESCRIPTION
**Pull request by Madhu-1**
# Describe what this PR does #

start controller or node server based on configuration, if both controller and node server flags are set/unset cephcsi will start both servers, if only one flag is set, it will start relevant service.

move flag configuration variable to util remove unwanted checks remove getting driver type from the binary name

rename rbd.go to driver.go  to keep code struct similar with cephfs  renamed rbd.go to driver.go

## Is there anything that requires special attention ##

Do you have any questions?

Is the change backward compatible? yes 

Fixes: #271 

## Future concerns ##

List items that are not part of the PR and do not impact it's
functionality, but are work items that can be taken up subsequently.

### CODE REVIEW

**Comment by Madhu-1** [comment]
@humblec @ShyamsundarR @poornimag PTAL

File: pkg/cephfs/driver.go [hunk 1]
```diff
@@ -147,20 +147,30 @@ func (fs *Driver) Run(conf *util.Config, cachePersister util.CachePersister) {
...
 
-	fs.cd.AddControllerServiceCapabilities([]csi.ControllerServiceCapability_RPC_Type{
-		csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME,
-	})
-
-	fs.cd.AddVolumeCapabilityAccessModes([]csi.VolumeCapability_AccessMode_Mode{
-		csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER,
-	})
-
+	if conf.IsControllerServer || !(conf.IsControllerServer && conf.IsNodeServer) {
```
Comment (ShyamsundarR) [inline pkg/cephfs/driver.go]: If isNS is `true`and isCS is `false` we would still go ahead and add the capability information as the `!(isCS && isNS)` evaluates to true, similarly in line 170, if only one of the services are true, then both services would be started.

I think the code intends to do the following in line 150,
```
if isCS || !isNS
```

and in line 170 this,
```
if !isCS && !isNS
```

Comment (Madhu-1) [reply pkg/cephfs/driver.go]: yes, updated code now

**Comment by humblec** [comment]
@Madhu-1 please rebase. We will get this in `1.2.0`.

**Comment by Madhu-1** [comment]
@humblec done

**Pull request merged**